### ISSUE TITLE
JSON.pretty_generate generates wrong format json output to MRI for at least an empty hash version 9.4.0.0

### ISSUE THREAD

**Issue by tqtu** [flori/json#515]
Code:

```
require 'json'
puts "With empty hash"
puts JSON.pretty_generate({})
puts "With empty hash in child"
puts JSON.pretty_generate({"empty": {}})

```
Output: jruby -v hello.rb
```
jruby 9.4.0.0 (3.1.0) 2022-11-23 95c0ec159f Java HotSpot(TM) 64-Bit Server VM 25.202-b08 on 1.8.0_202-b08 +jit [x86_64-darwin]
With empty hash
{
}
With empty hash in child
{
  "empty": {
}
}
```
The indentation is incorrect.
CC: @headius

**Comment by headius** [comment]
Thank you for the report!

**Comment by headius** [comment]
Likely caused by changes in https://github.com/flori/json/pull/449.

**Comment by headius** [comment]
Fix in #517.

**Comment by mrckzgl** [comment]
We experience the empty hash problem also with MRI ruby `ruby 3.2.1 (2023-02-08 revision 31819e82c8) [x86_64-linux]` while using the `json/pure` variant. Should I open a new issue for this?
